plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadataFactory.java

```diff
@@ -29,28 +29,29 @@ public class DefaultJdbcMetadataFactory
     private final JdbcClient jdbcClient;
     private final TimestampTimeZoneDomain timestampTimeZoneDomain;
     private final Set<JdbcQueryEventListener> jdbcQueryEventListeners;
+    private final IdentityCacheMapping identityCacheMapping;
 
     @Inject
     public DefaultJdbcMetadataFactory(
             JdbcClient jdbcClient,
             TimestampTimeZoneDomain timestampTimeZoneDomain,
-            Set<JdbcQueryEventListener> jdbcQueryEventListeners)
+            Set<JdbcQueryEventListener> jdbcQueryEventListeners,
+            IdentityCacheMapping identityCacheMapping)
```

Comment:
@kokosing @Praveen2112 I'm wondering if it has to be annotated as @ForMetadataCache? As there may be different caches with different strategies.

Reply:
But I guess the caching eviction/ cache key generation should be based on injected `IdentityCacheMapping`